versioned/tiered/rocksdb/src/main/java/org/projectnessie/versioned/rocksdb/RocksFragment.java

```diff
@@ -127,4 +139,77 @@ public void evaluate(Function function) throws ConditionFailedException {
   private Entity keysAsEntityList(List<Key> keys) {
     return Entity.ofList(keys.stream().map(k -> Entity.ofList(k.getElements().stream().map(Entity::ofString))));
   }
+
+  @Override
+  protected void remove(String fieldName, ExpressionPath.PathSegment path) {
+    if (new PathPattern().anyPosition().matches(path)) {
+      final int i = path.asPosition().getPosition();
+      fragmentBuilder.removeKeys(i);
+    } else if (new PathPattern().anyPosition().anyPosition().matches(path)) {
+      final int outerIndex = path.asPosition().getPosition();
+      final int innerIndex = path.getChild().get().asPosition().getPosition();
+
+      final List<String> updatedKeys = new ArrayList<>(fragmentBuilder.getKeys(outerIndex).getElementsList());
+      updatedKeys.remove(innerIndex);
+
+      fragmentBuilder.setKeys(path.asPosition().getPosition(), ValueProtos.Key
+          .newBuilder()
+          .addAllElements(updatedKeys)
+      );
+    } else {
+      throw new UnsupportedOperationException("Invalid path for remove");
+    }
+  }
+
+  @Override
+  protected boolean fieldIsList(String fieldName, ExpressionPath.PathSegment childPath) {
+    if (!KEY_LIST.equals(fieldName)) {
+      return false;
+    }
+
+    return new PathPattern().matches(childPath) || new PathPattern().anyPosition().matches(childPath);
+  }
+
+  @Override
+  protected void appendToList(String fieldName, ExpressionPath.PathSegment childPath, List<Entity> valuesToAdd) {
+    if (new PathPattern().matches(childPath)) {
```

Comment:
This looks odd, why isn't matches a static?

Reply:
I can see that you would like to remove the PathPattern class as you don't see any benefit to it.
If I remove it, then I will be dealing with **ExpressionPath.PathSegment** and I will receive a lot of feedback about how difficult it is to understand what is going on.

new PathPattern().nameEquals("foo").anyPosition() is equivalent to:
if path != null
  && path.isName()
  && "foo".equals(path.asName().getName())
  && path.getChild().isPresent()
  && path.getChild().get().isPosition())

Reply:
It's not that I want to have it removed, just that it looks odd to have to new it every time it's used. I wonder if it wouldn't be better to have a static builder or something?

Reply:
This could be improved, and we could have the path patterns stored static in the classes. I'll leave as is for now, since you likely have a different approach in mind.

Reply:
The PathPattern has been updated. It is still created new each time it is used. The next step would be to extract the PathPatterns into static variables.